sdk/core/core/src/az_span.c

```diff
@@ -46,7 +46,13 @@ AZ_NODISCARD az_span az_span_slice(az_span span, int32_t low_index, int32_t high
 
 AZ_NODISCARD AZ_INLINE uint8_t az_ascii_lower(uint8_t value)
 {
-  return 'A' <= value && value <= 'Z' ? value + AZ_ASCII_LOWER_DIF : value;
+  // This is equivalent to the following but with fewer conditions.
+  // return 'A' <= value && value <= 'Z' ? value + AZ_ASCII_LOWER_DIF : value;
+  if ((uint8_t)(int8_t)(value - 'A') <= ('Z' - 'A'))
+  {
+    value += AZ_ASCII_LOWER_DIF;
```

Comment:
We are probably not consistent on this yet, but I freaked out a bit when I saw the name `AZ_ASCII_` as if it is a part of our public interface. It is not, which is good news - it is defined right in this .c file and is only visible there, but can you please rename it to `_az_ASCII_LOWER_DIF`?